clang/include/clang/Basic/Attr.td

```diff
@@ -2020,6 +2020,25 @@ def : MutualExclusions<[SYCLIntelFPGAIVDep,
 def : MutualExclusions<[SYCLIntelFPGAMaxConcurrency,
                         SYCLIntelFPGADisableLoopPipelining]>;
 
+def SYCLIntelFpgaPipeline : InheritableAttr {
+  let Spellings = [CXX11<"intel","fpga_pipeline">];
+  let Args = [ExprArgument<"Value", /*optional*/1>];
+  let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let Subjects = SubjectList<[ForStmt, CXXForRangeStmt, WhileStmt, DoStmt, Function],
+                              ErrorDiag,
+                              "'for', 'while', 'do' statements, and functions">;
+  let Documentation = [SYCLIntelFpgaPipelineAttrDocs];
+  let IsStmtDependent = 1;
+  let SupportsNonconformingLambdaSyntax = 1;
```

Comment:
Btw, I think you should push back with the feature author on this point.

Reply:
I will check with author. Thanks Aaaron.

Reply:
I checked with author:

```
Motivation:
1.attribute disable_loop_pipelining was designed that it doesn't take any argument. What this attribute does today is that it can only disable pipelining. However, the main idea behind the fpga_pipeline attribute is to allow users to enable pipelining. 
2.the naming of disable_loop_pipelining might be misleading when applied to a function. 
3.it seems weird to have two different attributes that one enables pipelining and one disables it, so we might want just create a unified one with a better naming.
```
The attributes needs to apply on lambda function for IP Authoring work.